### PR TITLE
HIVE-24871: Initiator / Cleaner performance metrics

### PR DESCRIPTION
**Pull request by deniskuzZ**
### What changes were proposed in this pull request?

Introduced performance metrics for Initiator and Cleaner

### Why are the changes needed?

Compaction observability

### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

Unit tests

### CODE REVIEW

File: ql/src/java/org/apache/hadoop/hive/ql/txn/compactor/Initiator.java [hunk 1]
```diff
@@ -171,9 +176,12 @@ public void run() {
               StringUtils.stringifyException(t));
         }
         finally {
-          if(handle != null) {
+          if (handle != null) {
             handle.releaseLocks();
           }
+          if (metricsEnabled) {
```
Comment (pvargacl) [inline ql/src/java/org/apache/hadoop/hive/ql/txn/compactor/Initiator.java:182]: Won't this fail, if the acquireLock times out and the perflogger was not started? The PerfLogger in ql has a method for checking this startTimeHasMethod, maybe it worth to copy that and use it here

File: ql/src/test/org/apache/hadoop/hive/ql/txn/compactor/TestInitiator.java [hunk 2]
```diff
@@ -44,11 +44,7 @@
 import org.junit.Assert;
 import org.junit.Test;
 
-import java.util.ArrayList;
-import java.util.Collections;
-import java.util.HashMap;
-import java.util.List;
-import java.util.Map;
+import java.util.*;
```
Comment (pvargacl) [inline ql/src/test/org/apache/hadoop/hive/ql/txn/compactor/TestInitiator.java:47]: revert this please

File: ql/src/java/org/apache/hadoop/hive/ql/txn/compactor/Initiator.java (hunk 1 above)
Comment (deniskuzZ) [reply ql/src/java/org/apache/hadoop/hive/ql/txn/compactor/Initiator.java:182]: checked: it won't fail and won't report metric in this case

File: ql/src/test/org/apache/hadoop/hive/ql/txn/compactor/TestInitiator.java (hunk 2 above)
Comment (deniskuzZ) [reply ql/src/test/org/apache/hadoop/hive/ql/txn/compactor/TestInitiator.java]: fixed

**Review by pvargacl — APPROVED** [review]
LGTm +1 depending tests

**Pull request merged**